docs/cs-proof.md

```diff
@@ -609,6 +609,115 @@ to prove that
 \\(t(x) = {\langle {\mathbf{l}}(x), {\mathbf{r}}(x) \rangle}\\).
 
 
+Padding \\(\mathbf{l}(x)\\) and \\(\mathbf{r}(x)\\) for the inner product proof
+-------------------------------------------------------------------------------
+
+The above discussion did not have restrictions on the value \\(n\\).
+However, the [inner product argument](../notes/index.html#inner-product-proof)
+requires the input vectors to have power-of-two elements: \\(n^{+} = 2^k\\).
+To resolve this mismatch, we need to specify how to pad vectors \\(\mathbf{l}(x), \mathbf{r}(x)\\)
+and related commitments before we can use the inner product argument.
+
+Our goal is to translate the _padding of the constraint system_ into the _padding of proof data_,
+so we can keep the constraint system small and perform less computations in proving and verification.
+
+We start by padding the entire constraint system:
+multipliers are padded with all-zero assignments \\(a\_{L,j}, a\_{R,j}, a\_{O,j}\\),
+all-zero blinding factors \\(s\_{L,j}, s\_{R,j}\\),
+and all-zero weights \\(W\_{R,i,j}, W\_{L,i,j}, W\_{O,i,j}\\),
+for all constraints \\(i \in [0, q)\\) and all additional multipliers \\(j \in [n,n')\\):
+
+\\[
+\begin{aligned}
+\mathbf{a}\_L^{+} &= \mathbf{a}\_L \hspace{0.1cm} || \hspace{0.1cm} [0,...,0] \\\\
```

Comment:
Should we write this as `[0, ..., 0]` or as `\mathbf 0`?

Reply:
maybe \mathbf 0^{n^+ - n} (or whatever the length of the zero vector is for that instance) would be the most informative... right now, it's not super clear what its length is

Reply:
Yeah, i’ll introduce a `delta n` probably to make it all concrete.

Reply:
Does it matter what its length is?

Reply:
in some cases yes - such as for `l(x)+`, it would be nice to know how many zeroes the vector is padded with. (you can deduce from following the math or comparing it to `r(x)+`, but it seems nice to note the length explicitly)

Reply:
Hmm, I'm not sure I understand why you would need to know how many zeros the vector is padded with, and if you don't, it seems better not to give it a notation at all (to reduce cognitive overhead and hint that it's not important).

Reply:
Looks cleaner indeed:

<img width="957" alt="image" src="https://user-images.githubusercontent.com/698/46896787-eecf0a00-ce33-11e8-9281-c8712e7d1206.png">